### PR TITLE
feat: add BUNDLE metadata to allow Phygitals offers

### PR DESCRIPTION
**Pull request by levalleux-ludo**
## Description

{{what was done}}

## How to test

{{how can it be tested}}

### CODE REVIEW

File: packages/metadata/src/rNFT/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  type: "rNFT";
+  name: string;
+  description: string;
+  image?: string;
+  externalUrl: string;
+  licenseUrl: string;
+  condition?: string;
+  animationUrl?: string;
+  attributes?: {
+    trait_type: string;
```
Comment (albertfolch-redeemeum) [inline packages/metadata/src/rNFT/index.ts:21]: can we change it to?
```suggestion
    traitType: string;
```

File: packages/metadata/src/rNFT/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    value: string;
+    display_type?: string;
```
Comment (albertfolch-redeemeum) [inline packages/metadata/src/rNFT/index.ts:23]:
```suggestion
    displayType?: string;
```

File: packages/metadata/tests/productV1Item.test.ts [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      expect(variants.length).toBeGreaterThanOrEqual(4);
+      variants[variants.length - 2].productVariant = [
+        ...variants[1].productVariant
+      ];
+      expect(() => createVariantProductItem(productItem, variants)).toThrow(
+        `Redundant variant ${serializeVariant(variants[1].productVariant)}`
+      );
+    });
+
+    test("should NOT fail if the variants table is disordered ", async () => {
```
Comment (albertfolch-redeemeum) [inline packages/metadata/tests/productV1Item.test.ts:275]:
```suggestion
    test("should NOT fail if the variants table is unordered ", async () => {
```

(same in other places)

File: packages/metadata/src/rNFT/index.ts (hunk 2 above)
Comment (levalleux-ludo) [reply packages/metadata/src/rNFT/index.ts:23]: Done but had to keep compliance with NFT "standard" https://docs.opensea.io/docs/metadata-standards, at least for collection and nftItem

File: packages/metadata/src/rNFT/index.ts (hunk 1 above)
Comment (levalleux-ludo) [reply packages/metadata/src/rNFT/index.ts:21]: Done but had to keep compliance with NFT "standard" https://docs.opensea.io/docs/metadata-standards, at least for collection and nftItem

File: packages/metadata/tests/productV1Item.test.ts (hunk 3 above)
Comment (levalleux-ludo) [reply packages/metadata/tests/productV1Item.test.ts:275]: Done (all occurences)

File: packages/core-sdk/src/metadata/subgraph.ts [hunk 4]
```diff
@@ -256,13 +260,24 @@ export async function getProductWithVariants(
...
+      (item) =>
+        (item as ProductV1ItemMetadataEntity).productUuid === productUuid
+    )
+    .map((item) => {
+      return {
+        productV1Item: item as ProductV1ItemMetadataEntity,
+        offer: bundle.offer as ProductV1MetadataEntityFieldsFragment["offer"]
+      };
+    });
+}
```
Comment (albertfolch-redeemeum) [inline packages/core-sdk/src/metadata/subgraph.ts]: you can get rid of some castings by adding the "is <type>" in the filter function so Typescript knows better how to narrow down the type automatically
```
  return bundle.items
    .filter(
      (item): item is ProductV1ItemMetadataEntity =>
        item.type === ItemMetadataType.ItemProductV1
    )
    .filter((item) => item.productUuid === productUuid)
    .map((item) => {
      return {
        productV1Item: item,
        offer: bundle.offer as ProductV1MetadataEntityFieldsFragment["offer"]
      };
    });
```

**Review by albertfolch-redeemeum — APPROVED** [review]
e2e tests are failing

File: packages/core-sdk/src/metadata/subgraph.ts (hunk 4 above)
Comment (levalleux-ludo) [reply packages/core-sdk/src/metadata/subgraph.ts]: Done

**Comment by levalleux-ludo** [comment]
closes #611

**Pull request merged**